src/optimizer/filter_combiner.cpp

```diff
@@ -595,40 +596,110 @@ TableFilterSet FilterCombiner::GenerateTableScanFilters(const vector<idx_t> &col
-				if (in_values[in_val_idx] - in_values[in_val_idx - 1] > 1) {
-					can_simplify_in_clause = false;
-					break;
+				for (idx_t in_val_idx = 1; in_val_idx < in_values.size(); in_val_idx++) {
+					if (in_values[in_val_idx] - in_values[in_val_idx - 1] > 1) {
+						can_simplify_in_to_range = false;
+						break;
+					}
 				}
 			}
-			if (!can_simplify_in_clause) {
-				continue;
+			// if we are not consecutive or we are VARCHAR, then we can still push an OR zone_map filter
+			if (!can_simplify_in_to_range && type.IsIntegral()) {
+				auto or_filter = make_uniq<ConjunctionOrFilter>();
+				for (idx_t in_val_idx = 1; in_val_idx < func.children.size(); in_val_idx++) {
+					D_ASSERT(func.children[in_val_idx]->type == ExpressionType::VALUE_CONSTANT);
+					auto &const_val = func.children[in_val_idx]->Cast<BoundConstantExpression>();
+					auto zone_map_filter = make_uniq<ZoneMapFilter>();
+					zone_map_filter->child_filter =
+					    make_uniq<ConstantFilter>(ExpressionType::COMPARE_EQUAL, const_val.value);
+					or_filter->child_filters.push_back(std::move(zone_map_filter));
+				}
+				table_filters.PushFilter(column_index, std::move(or_filter));
+				break;
+			} else if (can_simplify_in_to_range && type.IsIntegral()) {
+				auto lower_bound = make_uniq<ConstantFilter>(ExpressionType::COMPARE_GREATERTHANOREQUALTO,
+				                                             Value::Numeric(type, in_values.front()));
+				auto upper_bound = make_uniq<ConstantFilter>(ExpressionType::COMPARE_LESSTHANOREQUALTO,
+				                                             Value::Numeric(type, in_values.back()));
+				table_filters.PushFilter(column_index, std::move(lower_bound));
+				table_filters.PushFilter(column_index, std::move(upper_bound));
+				table_filters.PushFilter(column_index, make_uniq<IsNotNullFilter>());
+
+				remaining_filters.erase_at(rem_fil_idx);
 			}
-			auto lower_bound = make_uniq<ConstantFilter>(ExpressionType::COMPARE_GREATERTHANOREQUALTO,
-			                                             Value::Numeric(type, in_values.front()));
-			auto upper_bound = make_uniq<ConstantFilter>(ExpressionType::COMPARE_LESSTHANOREQUALTO,
-			                                             Value::Numeric(type, in_values.back()));
-			table_filters.PushFilter(column_index, std::move(lower_bound));
-			table_filters.PushFilter(column_index, std::move(upper_bound));
-			table_filters.PushFilter(column_index, make_uniq<IsNotNullFilter>());
+		}
+	}
+
+	for (idx_t rem_fil_idx = 0; rem_fil_idx < remaining_filters.size(); rem_fil_idx++) {
+		auto &remaining_filter = remaining_filters[rem_fil_idx];
+		if (remaining_filter->expression_class == ExpressionClass::BOUND_CONJUNCTION) {
+			auto &conj = remaining_filter->Cast<BoundConjunctionExpression>();
+			if (conj.type == ExpressionType::CONJUNCTION_OR) {
+				bool column_id_set = false;
+				bool same_column_id = true;
+				idx_t column_id = 0;
```

Comment:
Can we use an `optional_idx column_id` instead of `bool column_id_set` and `idx_t column_id`? If unset (invalid), it is equal to `DConstants::INVALID_INDEX`.

In the current implementation this may cause a problem because it this is equal to `COLUMN_IDENTIFIER_ROW_ID`, which is also used for `IsRowIdColumnId`. However, if we immediately check if `IsRowIdColumnId(column_ids[column_ref->binding.column_index])` I don't think that should be a problem unless I'm missing something.